src/event/writer.rs

```diff
@@ -118,10 +123,22 @@ impl EventWriter {
     pub async fn write_event(&mut self, event: Vec<u8>) -> oneshot::Receiver<Result<(), Error>> {
         let size = event.len();
         let (tx, rx) = oneshot::channel();
+        let (tx_flush, rx_flush) = oneshot::channel();
         let routing_info = RoutingInfo::RoutingKey(None);
+        let routing_info_flush = RoutingInfo::RoutingKey(None);
         if let Some(pending_event) = PendingEvent::with_header(routing_info, event, None, tx) {
             let append_event = Incoming::AppendEvent(pending_event);
-            self.writer_event_internal(append_event, size, rx).await
+            let write_event = self.writer_event_internal(append_event, size, rx).await;
```

Comment:
Instead of calling writer_event_internal here, it might be better to modify it to have a single call which takes two `rx`es. (IE embedding both `tx`es into the same PendingEvent.)

This would avoid a bug which is currently in this code which is that it is passing `None` as the routing key for both the `rx_flush` and the `rx` which implies that it is legal for these to be sent to different segments.

Reply:
I added an additional optional Sender within PendingEvent, and write_event_internal now is only called once to track flush activity as well as write. This should eliminate our concerns dealing with different routing keys, and different segments.